diff.py

```diff
@@ -1609,6 +1674,62 @@ def _normalize_adrp_differences(self, mnemonic: str, row: str) -> str:
         return row
 
 
+class AsmProcessorI686(AsmProcessor):
+    def process_reloc(self, row: str, prev: str) -> Tuple[str, Optional[str]]:
+        arch = self.config.arch
+        repl = row.split()[-1]
+        args_string = ""
+        reverse = False
+        mnemonic, args = prev.split(maxsplit=1)
+
+        comma = args.find(",")
+        open_bracket = args.find("(")
+
+        # Register relative addressed target e.g.
+        # R_386_GOTOFF        x
+        # movl      $0x1,0x0(%eax)
+        if comma != -1:
+            if open_bracket != -1 and open_bracket > comma:
+                reverse = True
+                args_string = args[: comma + 1]
+            else:
+                args_string = args[comma:]
```

Comment:
This logic is incomplete:
```c
int x[2], y[2];

int foo() {
	y[1] = 1;
	return x[1];
}
```
compiled with `gcc test.c -c -m32 -O2 -fno-pic && objdump --no-show-raw-in -dr test.o` yields
```
00000000 <foo>:
   0:	movl   $0x1,0x4
			2: R_386_32	y
   a:	mov    0x4,%eax
			b: R_386_32	x
   f:	ret    
```

`movl   $0x1,0x4` has the relocated symbol as second argument, while this code assumes the first.

This also shows off an interesting property of GCC-compiled x86 ELFs, in that GCC chooses to use REL relocations, with addend embedded into the asm. (On x86_64 it uses RELA relocs where they are separate, resulting in the equivalent of `movl   $0x1,0x0` / `R_386_32   y+4` instead). I don't know how your toolchain looks for MSVC compilation but I imagine RELA is more likely so no need to worry about this I guess.

Also, I'm not sure stripping parentheses entirely is the right move. At least for MIPS it would result in invalid asm, and it loses information. Example:
```c
extern int x[20], y[20];

int foo(int a) {
	y[a + 1] = 1;
	return x[a + 2];
}
```

```
00000000 <foo>:
   0:	mov    0x4(%esp),%eax
   4:	movl   $0x1,0x4(,%eax,4)
			7: R_386_32	y
   f:	mov    0x8(,%eax,4),%eax
			12: R_386_32	x
  16:	ret    
```